blueocean-events/src/test/java/io/jenkins/blueocean/events/sse/SSEConnection.java

```diff
@@ -0,0 +1,267 @@
+package io.jenkins.blueocean.events.sse;
+
+import com.ning.http.client.AsyncHandler;
+import com.ning.http.client.AsyncHttpClient;
+import com.ning.http.client.AsyncHttpClient.BoundRequestBuilder;
+import com.ning.http.client.Cookie;
+import com.ning.http.client.HttpResponseBodyPart;
+import com.ning.http.client.HttpResponseHeaders;
+import com.ning.http.client.HttpResponseStatus;
+import com.ning.http.client.ListenableFuture;
+import com.ning.http.client.Response;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import org.jenkinsci.plugins.pubsub.ChannelSubscriber;
+import org.jenkinsci.plugins.pubsub.EventFilter;
+import org.jenkinsci.plugins.pubsub.Message;
+import org.jenkinsci.plugins.pubsub.SimpleMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+
+/**
+ * Represents an SSE connection to a server.
+ *
+ * @author Kohsuke Kawaguchi
+ * @author Vivek Pandey
+ * @see <a href="https://www.w3.org/TR/eventsource/">spec</a>
+ */
+public class SSEConnection implements Closeable {
```

Comment:
Yes, we did as part of blue steel poc. I changed it to use as test quality utility.